src/xunit.v3.core/Sdk/v3/Utility/FixtureMappingManager.cs

```diff
@@ -0,0 +1,236 @@
+		string fixtureCategory,
+		object[] cachedFixtureValues) :
+			this(fixtureCategory)
+	{
+		foreach (var cachedFixtureValue in Guard.ArgumentNotNull(cachedFixtureValues))
+			fixtureCache[cachedFixtureValue.GetType()] = cachedFixtureValue;
+	}
+
+	internal IReadOnlyDictionary<Type, object> FixtureCache => fixtureCache;
+
+	/// <summary>
+	/// Registers a type as a supported type for this level of fixtures. Any request to <see cref="GetFixture(Type)"/>
+	/// that isn't for one of the registered types will be forwarded onto the parent mapping manger, if one was provided.
+	/// Note that you can register open generic types here, and the system will be able to provide any instance of a
+	/// closed generic version of such types.
+	/// </summary>
+	/// <param name="type">The registered type</param>
+	[Obsolete("use initialize instead")]
+	public void AddFixtureType(Type? type)
+	{
+		if (type is not null)
+			knownTypes.Add(type);
+	}
+
+	/// <inheritdoc/>
+	public async ValueTask DisposeAsync()
+	{
+		if (disposed)
+			return;
+
+		GC.SuppressFinalize(this);
+		disposed = true;
+
+		var aggregator = new ExceptionAggregator();
+
+		var disposeAsyncTasks =
+			fixtureCache
+				.Values
+				.OfType<IAsyncDisposable>()
+				.Select(fixture => aggregator.RunAsync(async () =>
+				{
+					try
+					{
+						await fixture.DisposeAsync();
+					}
+					catch (Exception ex)
+					{
+						throw new TestFixtureCleanupException(string.Format(CultureInfo.CurrentCulture, "{0} fixture type '{1}' threw in DisposeAsync", fixtureCategory, fixture.GetType().FullName), ex.Unwrap());
+					}
+				}).AsTask())
+				.ToList();
+
+		if (disposeAsyncTasks.Count != 0)
+			await Task.WhenAll(disposeAsyncTasks);
+
+		foreach (var fixture in fixtureCache.Values.OfType<IDisposable>())
+			aggregator.Run(() =>
+			{
+				try
+				{
+					fixture.Dispose();
+				}
+				catch (Exception ex)
+				{
+					throw new TestFixtureCleanupException(string.Format(CultureInfo.CurrentCulture, "{0} fixture type '{1}' threw in Dispose", fixtureCategory, fixture.GetType().FullName), ex.Unwrap());
+				}
+			});
+
+		aggregator.ThrowIfFaulted();
+	}
+
+	/// <summary>
+	/// Get a value for the given fixture type. If the fixture type is unknown, then returns <c>null</c>.
+	/// </summary>
+	/// <param name="fixtureType">The type of the fixture</param>
+	/// <returns>Returns the value if the fixture type is found, or <c>null</c> if it's not.</returns>
+	public async ValueTask<object?> GetFixture(Type fixtureType)
+	{
+		if (disposed)
+			throw new ObjectDisposedException(nameof(FixtureMappingManager));
+
+		Guard.ArgumentNotNull(fixtureType);
+
+		// Pull from the cache if present
+		if (fixtureCache.TryGetValue(fixtureType, out var result))
+			return result;
+
+		// Ensure this is a type that's known to come from this fixture level; otherwise, ask the
+		// parent mapping manager to generate the type (or return null if it's not)
+		var knownType = knownTypes.Contains(fixtureType);
```

Comment:
@bradwilson Later in Initialize, "knownType" has type Type, which makes sense, but here refers to a boolean. Perhaps this one should be renamed containsKnownType or knownTypeFound?